src/scales/scale.time.js

```diff
@@ -360,8 +360,9 @@ module.exports = function(Chart) {
 		getLabelForIndex: function(index, datasetIndex) {
 			var me = this;
 			var label = me.chart.data.labels && index < me.chart.data.labels.length ? me.chart.data.labels[index] : '';
+			var firstData = me.chart.data.datasets[datasetIndex].data[index];
 
-			if (typeof me.chart.data.datasets[datasetIndex].data[0] === 'object') {
+			if (firstData !== null && typeof firstData === 'object') {
 				label = me.getRightValue(me.chart.data.datasets[datasetIndex].data[index]);
```

Comment:
@simonbrunel, agreed, it looks more condensed and readable. 
i'll make that change now.